apps/comps/app/competitions/[id]/og-image/route.tsx

```diff
@@ -0,0 +1,302 @@
+  if (!cachedFontsPromise) {
+    cachedFontsPromise = (async () => {
+      const [
+        geistLight,
+        geistRegular,
+        geistBold,
+        robotoMonoRegular,
+        robotoMonoBold,
+      ] = await Promise.all([
+        loadGoogleFont("Geist", 300),
+        loadGoogleFont("Geist", 400),
+        loadGoogleFont("Geist", 700),
+        loadGoogleFont("Roboto Mono", 400),
+        loadGoogleFont("Roboto Mono", 700),
+      ]);
+
+      return [
+        { name: "Geist", data: geistLight, weight: 300, style: "normal" },
+        { name: "Geist", data: geistRegular, weight: 400, style: "normal" },
+        { name: "Geist", data: geistBold, weight: 700, style: "normal" },
+        {
+          name: "Roboto Mono",
+          data: robotoMonoRegular,
+          weight: 400,
+          style: "normal",
+        },
+        {
+          name: "Roboto Mono",
+          data: robotoMonoBold,
+          weight: 700,
+          style: "normal",
+        },
+      ];
+    })();
+  }
+  return cachedFontsPromise;
+}
+
+/**
+ * Loads an asset from the public directory and returns it as a base64 data URL.
+ *
+ * @param assetPath - Path relative to the public directory
+ * @returns Base64 data URL for the asset, or null if the asset cannot be loaded
+ */
+async function loadAssetAsBase64(assetPath: string): Promise<string | null> {
+  try {
+    const fullPath = join(process.cwd(), "public", assetPath);
+    const buffer = await readFile(fullPath);
+    const base64 = buffer.toString("base64");
+
+    const ext = extname(assetPath).toLowerCase();
+    let mimeType = "image/svg+xml";
+    if (ext === ".png") mimeType = "image/png";
+    if (ext === ".jpg" || ext === ".jpeg") mimeType = "image/jpeg";
+
+    return `data:${mimeType};base64,${base64}`;
+  } catch (err) {
+    console.error(`Failed to load asset at path "${assetPath}":`, err);
+    return null;
+  }
+}
+
+/**
+ * Loads static assets for OG image generation.
+ * Asset data is cached in memory to avoid repeated file system reads on each OG image generation.
+ *
+ * @returns Object containing base64-encoded assets
+ */
+async function loadAssets(): Promise<CachedAssets> {
+  if (!cachedAssetsPromise) {
+    cachedAssetsPromise = (async () => {
+      const [backgroundImage, recallTokenSvg, recallLogoSvg] =
+        await Promise.all([
+          loadAssetAsBase64("og-background.png"),
+          loadAssetAsBase64("recall-token.svg"),
+          loadAssetAsBase64("logo_full_grey.svg"),
+        ]);
+      return { backgroundImage, recallTokenSvg, recallLogoSvg };
+    })();
+  }
+  return cachedAssetsPromise;
+}
+
+export async function GET(
+  _req: Request,
+  context: { params: Promise<{ id: string }> },
+): Promise<ImageResponse> {
+  const client = await createSafeClient();
+  const { id } = await context.params;
+
+  const [{ data: competition }, fonts, assets] = await Promise.all([
+    client.competitions.getById({ id }),
+    loadFonts(),
+    loadAssets(),
+  ]);
```

Comment:
What happens if any of these promises reject? I'm guessing a 500 error is returned to the client. Similar to how you return a default below if certain data isn't present, this error should be caught and the default image returned I think.